packages/discount-code-importer/src/main.js

```diff
@@ -0,0 +1,245 @@
+      error: npmlog.error.bind(this, ''),
+      warn: npmlog.warn.bind(this, ''),
+      info: npmlog.info.bind(this, ''),
+      verbose: npmlog.verbose.bind(this, ''),
+    }
+    this._resetSummary()
+  }
+
+  // Use static method because this is not called on any instance
+  static _buildPredicate (codeObjects: CodeDataArray): string {
+    const predicateArray = _.map(codeObjects, 'code')
+    return `code in ("${predicateArray.join('", "')}")`
+  }
+
+  // Wrapper function for compatibility with CLI
+  processStream (chunk: ChunkOptions, cb: () => mixed) {
+    this.logger.info('Starting conversion')
+    return this._processBatches(chunk)
+      .then(() => cb())
+      // No catch block as errors will be caught in the CLI
+  }
+
+  // Public function for direct module usage
+  run (codes: CodeDataArray) {
+    return this._processBatches(codes)
+  }
+
+  _processBatches (codes: CodeDataArray) {
+    // Batch to `batchSize` to reduce necessary fetch API calls
+    const batchedList = _.chunk(codes, this.batchSize)
+    return Promise.mapSeries(batchedList, (codeObjects: CodeDataArray) => {
+      // Build predicate and fetch existing code
+      const predicate = DiscountCodeImport._buildPredicate(codeObjects)
+      const service = this._createService()
+      const uri = service
+        .where(predicate)
+        .perPage(this.batchSize)
+        .build()
+      return this.client.execute({
+        uri,
+        method: 'GET',
+      })
+        .then(({ body: { results: existingCodes } }: Object) => (
+          this._createOrUpdate(codeObjects, existingCodes)
+        ))
+    })
+      .then(() => Promise.resolve(this.summaryReport()))
+      .catch(error => Promise.reject({
+        error: error.message ? error.message : error,
+        summary: this._summary,
+      }),
+      )
+  }
+
+  _createOrUpdate (newCodes: CodeDataArray, existingCodes: CodeDataArray) {
+    return Promise.map(newCodes, (newCode: CodeData) => {
+      const existingCode = _.find(existingCodes, ['code', newCode.code])
+      if (existingCode)
+        return this._update(newCode, existingCode)
+          .then((response) => {
+            if (response && response.statusCode === 304)
+              this._summary.unchanged += 1
+            else
+              this._summary.updated += 1
+            return Promise.resolve()
+          })
+          .catch((error) => {
+            if (this.continueOnProblems) {
+              this._summary.updateErrorCount += 1
+              this._summary.errors.push(
+                error.message ? error.message : error,
+              )
+              // eslint-disable-next-line max-len
+              const msg = 'Update error occured but ignored. See summary for details'
+              this.logger.error(msg)
+              return Promise.resolve()
+            }
+            // eslint-disable-next-line max-len
+            const msg = 'Process stopped due to error while creating discount code. See summary for details'
```

Comment:
I'm seeing a lot of `// eslint-disable-next-line max-len` here. Can we start looking into using [common-tags](https://github.com/declandewet/common-tags)?